agent/wsclient/client.go

```diff
@@ -230,15 +223,49 @@ func (cs *ClientServerImpl) IsReady() bool {
 
 // SetConnection passes a websocket connection object into the client. This is used only in
 // testing and should be avoided in non-test code.
-func (cs *ClientServerImpl) SetConnection(conn WebsocketConn) {
+func (cs *ClientServerImpl) SetConnection(conn wsconn.WebsocketConn) {
 	cs.conn = conn
 }
 
 // SetReadDeadline sets the read deadline for the websocket connection
 // A read timeout results in an io error if there are any outstanding reads
 // that exceed the deadline
 func (cs *ClientServerImpl) SetReadDeadline(t time.Time) error {
-	return cs.conn.SetReadDeadline(t)
+	err := cs.conn.SetReadDeadline(t)
+	if err == nil {
+		return nil
+	}
+	seelog.Warnf("Unable to set read deadline for websocket connection: %v for %s", err, cs.URL)
+	// If we get connection closed error from SetReadDeadline, break out of the for loop and
+	// return an error
+	if opErr, ok := err.(*net.OpError); ok && strings.Contains(opErr.Err.Error(), errClosed) {
```

Comment:
Should we do a `cs.Close()` somewhere in this case too?

Reply:
I don't think so since it's already closed.